model/metrics/database-metrics.yaml

```diff
@@ -8,9 +8,9 @@ groups:
     stability: experimental
     extends: attributes.db.client
 
-  - id: metric.db.client.connections.usage
```

Comment:
First time doing such a PR, so I rather ask here: shouldn't a deprecated metric be added with the old name? 

I see [it's done for attributes](https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/deprecated/db.yaml), but I don't see any for metrics - closest is [metric related attributes](https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/deprecated/db.yaml#L88), but that's still not the metric itself. Also looked at some similar PRs that rename a metric, and from what I have seen, the old metric name never shows up as deprecated.

Reply:
According to `make compatibility-check` and [CONTRIBUTING.md#version-compatibility-check](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#version-compatibility-check), `metric.db.client.connections.usage` still needs to show up.

I don't really find any example for deprecated metrics 🤔 - I added it to the end of `database-metrics.yaml`. Let me know if there is a better place for it.

Reply:
good question, i'm not sure we've done this before either, let's see what @lmolkova thinks

Reply:
we never deprecated metrics so far (which was not great and we started enforcing it very recently). So it's the first metrics to be deprecated.

Let's create `/model/metrics/deprecated/database.yaml` file and add deprecated metric there?

Reply:
Thanks @lmolkova - moved deprecated metrics to `/model/metrics/deprecated/database.yaml`. Should I also generate an `.md` file from that?

Reply:
@jsuereth I remember you wanted to have MD for all deprecated attributes to be generated in the registry.
Do you have a strong opinion for metrics?

I have a soft preference to not generate experimental-deprecated metrics in the MD to reduce noise while keeping codegen intact.

Reply:
based on the offline discussion, we decided to skip md for deprecated metrics now and add it later if it's ever necessary